File: student-service/src/main/java/com/vidyasampadana/student_service/service/StudentService.java
package com.vidyasampadana.student_service.service;


import com.vidyasampadana.student_service.entity.Students;

import java.util.List;

public interface StudentService {

    Students createStudent(Students student);

    List<Students> getAllStudents();

    Students getStudentById(Long id);

    Students updateStudent(Long id, Students updateStudent);

    void deleteStudent(Long id);
}

File: frontend/frontend/src/services/studentService.js
import API from "./api";

const API_URL = "/student";

export const getStudents = () => API.get(API_URL);

export const createStudent = (student) => API.post(API_URL, student);

export const updateStudent = (id, student) => API.put(`${API_URL}/${id}`, student);

export const deleteStudent = (id) => API.delete(`${API_URL}/${id}`);

export const getStudentById = (id) => {
  return API.get(`${API_URL}/${id}`);
};